enterprise/server/backends/memcache/memcache.go

```diff
@@ -34,6 +41,23 @@ type Cache struct {
 	prefix string
 }
 
+func Register(env *real_environment.RealEnv) error {
+	if fmt.Sprintf("%T", env.GetCache()) == fmt.Sprintf("%T", composable_cache.NewComposableCache(nil, nil, 0)) {
+		// Cache has already been composed, don't do it again.
```

Comment:
should this log a warning? does this indicate a misconfiguration or logic error in our code?

Reply:
Previously, we allowed both memcache and redis cache to be configured and memcache would silently take precedence if present, so these checks are intended to just duplicate that behavior by only allowing one composable cache to be used and, using the first one we try to register that is also configured. If we think this is confusing and want to log a warning, I have no objections, but it is a departure from the current way the system works.